docs/algorithms/string-algorithms/sbom-algorithm.md

```diff
@@ -0,0 +1,181 @@
+            oracle[i] = State();
+            oracle[i].length = i;
+        }
+        
+        // Build transitions
+        for (int i = 0; i < pattern.length(); i++) {
+            oracle[i].transitions[pattern[i]] = i + 1;
+            
+            // External transitions
+            for (int j = i - 1; j >= 0; j--) {
+                char c = pattern[i];
+                if (oracle[j].transitions.count(c)) {
+                    break;
+                }
+                oracle[j].transitions[c] = i + 1;
+            }
+        }
+    }
+
+public:
+    vector<int> search(const string& text, const string& pat) {
+        vector<int> matches;
+        pattern = pat;
+        
+        if (pattern.empty() || text.empty()) return matches;
+        
+        // Build oracle
+        buildOracle();
+        
+        // Searching phase
+        int pos = pattern.length() - 1;
+        while (pos < text.length()) {
+            int state = 0;
+            int j = pos;
+            
+            // Backward matching
+            while (j >= pos - pattern.length() + 1 && 
+                   oracle[state].transitions.count(text[j])) {
+                state = oracle[state].transitions[text[j]];
+                j--;
+            }
+            
+            // Check if match found
+            if (j == pos - pattern.length()) {
+                matches.push_back(pos - pattern.length() + 1);
+            }
+            
+            // Shift pattern
+            pos += pattern.length();
+        }
+        
+        return matches;
+    }
+    
+    // Advanced search with optimizations
+    vector<int> advancedSearch(const string& text, const string& pat) {
+        vector<int> matches;
+        pattern = pat;
+        
+        if (pattern.empty() || text.empty()) return matches;
+        
+        // Build oracle
+        buildOracle();
+        
+        // Precompute pattern characters
+        vector<bool> patternChars(256, false);
+        for (char c : pattern) {
+            patternChars[c] = true;
+        }
+        
+        // Searching with optimizations
+        int pos = pattern.length() - 1;
+        while (pos < text.length()) {
+            // Quick character check
+            if (!patternChars[text[pos]]) {
+                pos += pattern.length();
+                continue;
+            }
+            
+            int state = 0;
+            int j = pos;
+            
+            // Backward matching with early termination
+            while (j >= pos - pattern.length() + 1 && 
+                   oracle[state].transitions.count(text[j])) {
+                state = oracle[state].transitions[text[j]];
+                j--;
+            }
+            
+            if (j == pos - pattern.length()) {
+                matches.push_back(pos - pattern.length() + 1);
+            }
+            
+            pos += pattern.length();
+        }
+        
+        return matches;
+    }
+};
```

Comment:
close your code example with ```` ``` ````

Reply:
@ajay-dhangar There is a problem with repo . It is causing sync and update problems . Kindly, fix this